api/api-doc/storage_service.json

```diff
@@ -1122,6 +1122,14 @@
                      "allowMultiple":false,
                      "type":"string",
                      "paramType":"query"
+                  },
+                  {
+                     "name":"small_table_optimization",
+                     "description":"If the value is the string 'true' with any capitalization, perform small table optimization.",
```

Comment:
We need better documentation than this - either here or in a document describing repair. Nobody will ever know what "small table optimization" means, how small it needs to be, and what else this "optimization" assumes. I see you said above that this option not only assumes a "small table" but also means you don't need to run repair on all nodes, running it on one node is enough. This is completely unobvious from the name "small table optimization". So we need documentation (and no, the commit message isn't enough).

Reply:
Yes, we definitely need more documents about this. The patch focuses on providing the minimal core api. How user consumes the api is not solved here.  However, end user is not supposed to use the option at all. The option will be consumed either by nodetool or scylla manager (preferred). There is a big shift how user of this api should use it because it is a cluster wide repair.